charts/virtual-kubelet/templates/deployment.yaml

```diff
@@ -77,7 +79,8 @@ spec:
           "--namespace", "{{ .Values.monitoredNamespace }}",
           "--nodename", "{{ required "nodeName is required" .Values.nodeName }}",
           "--os", "{{ .Values.nodeOsType }}",
-          "--taint", "{{ .Values.nodeTaint }}"
+          "--taint", "{{ .Values.nodeTaint }}",
+          "--disable-default-taint", "{{ .Values.disableDefaultTaint }}"
```

Comment:
I'd be in favor a simpler approach where we only go with something like `virtual-kubelet.github.io:<provider>=<Effect>` as the built in Virtual Kubelet functionality and not include the `disable-default-taint` functionality.  I thought one of the goals here was to kind of harmonize on behavior that would allow multiple providers, but also the "don't care" behavior. I think adding the `disable-default-taint` behavior kind of distracts from that. Same for adding custom taints beyond the "default". It's fairly easy to add/remove whatever taints you want via kubectl, so if you want to deviate from the built in functionality, that's possible. (counter arguement: you could do that with the taints in general, but I feel like having the common format one makes VK a more usable thing across providers when i'm the end user).

Reply:
I was trying to think of a use case for needing two node taints, and have been coming up empty.

How about three environment variables:
- `VKUBELET_TAINT_KEY` (default `virtual-kubelet.github.io/provider`)
- `VKUBELET_TAINT_VALUE` (default `{{ .Values.provider }}`)
- `VKUBELET_TAINT_EFFECT` (default `NoSchedule`)

I'd be in favor of making this a breaking change in the helm values file so that those who have been overriding `azure.com/aci` get the new default taint key instead.

Reply:
One other thought is that we should probably provide a way to disable taints altogether on virtual-kubelet nodes.  That could be a future change too, but something to keep in mind for a time when vk is stable enough for users to not care about explicitly "opting in".  Perhaps that makes sense as a flag?

Reply:
Yeah, this is a great point ☝️ would, 100% be in favor of that. 

I'm good with either flags or environment variables for either of these, I just want to be consistent within the taint definition part..so either all envs (which works well with pod presets like you mentioned) or all flags.

Reply:
I like those three environment variables BTW, I vote this way. @rbitia @robbiezhang, any input?